torch/_dynamo/backends/debugging.py

```diff
@@ -132,3 +132,64 @@ def non_leaf_compile_error_TESTING_ONLY(gm: torch.fx.GraphModule, example_inputs
         if not t.is_leaf:
             raise TestingOnlyCompileError()
     return gm
+
+
+class ExplainWithBackend:
+    """
+    This class is intended to be used as a backend for `torch.compile`. It is
+    composable with other backends. When used in this way, it accumulates
+    information about graph breaks, ops, and other info and provides a string
+    representation summarizing this information.
+
+    Attributes:
+        backend (str): The name of the backend to use for optimization.
+        graphs (list): A list of the graphs captured by TorchDynamo.
+        op_count (int): The total number of operations in all optimized graphs.
+        break_reasons (list): A list of graph break reasons with stack traces.
+
+    Example Usage:
+        def fn(x):
+            x = torch.sigmoid(x)
+            return x
+
+        torch._dynamo.reset()
+        eb = ExplainWithBackend("inductor")
+        optimized_fn = torch.compile(fn, backend=eb)
+        result = optimized_fn(torch.randn(5))
+        print(eb)
+    """
+
+    def __init__(self, backend):
+        self.backend = backend
+        self.graphs = []
+        self.op_count = 0
+        self.break_reasons = []
+
+    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+        from .registry import lookup_backend
+
+        self.graphs.append(gm)
+        ops = []
+        for node in gm.graph.nodes:
+            if node.op == "call_function":
+                ops.append(node.target)
+        self.op_count += len(ops)
+        if gm.compile_subgraph_reason.graph_break:
+            self.break_reasons.append(gm.compile_subgraph_reason)
+
+        return lookup_backend(self.backend)(gm, example_inputs)
+
+    def __str__(self):
```

Comment:
Not bad, but I want to see you utilize the structure @msaroufim added here https://github.com/pytorch/pytorch/pull/102869/files#diff-c663df72d7a648204ff256e604fe3c9b393e7e2d58204e3bc818d90a99712911R558

Reply:
Updated the structure and made sure it's used in both places (y)